broker/service/api/__init__.py

```diff
@@ -130,14 +130,11 @@ def get_node_cluster(k8s_conf_path):
     try:
         kube.config.load_kube_config(k8s_conf_path)
         CoreV1Api = kube.client.CoreV1Api()
-
-        node_number = len(CoreV1Api.list_node().items)
-
-        node_info = CoreV1Api.list_node().items[node_number - 1]
+        for node in CoreV1Api.list_node().items:
+            is_ready = [s for s in node.status.conditions if s.type == 'Ready'][0].status == 'True'
```

Comment:
`CoreV1Api.list_node()` has a parameter called `field_selector` which supports a few queries to filter data, but they don't seem to support the field `conditions` yet. IMO implementing that (like you did here) would be a great contribution for the python-kubernetesclient codebase :-)